src/device_utils.cpp

```diff
@@ -36,10 +37,16 @@ str_map get_serial_dev_info()
         "usb",
         "usb_device");
 
-      if (parent_dev)
+      if (parent_dev != nullptr)
       {
-        std::string dev_serial = std::string(udev_device_get_sysattr_value(parent_dev, "serial"));
-        devices[dev_file_name_path] = dev_serial;
+        const auto serial = udev_device_get_sysattr_value(parent_dev, "serial");
+        if (serial) {
```

Comment:
do we need to do != nullptr check here?

Reply:
We shouldn't need to do that at all with C++11, but I don't think it hurts too much to have it.